common/model/db/tables/node/select.js

```diff
@@ -0,0 +1,73 @@
+import * as Node from '../../../../../core/record/node'
+import * as Schemata from '../../schemata'
+
+/**
+ * Generates the select query for the node table by the given parameters.
+ *
+ * @param {!object} params - The query parameters.
+ * @param {string} [params.uuid=null] - The node uuid to filter by.
+ * @param {string} [params.recordUuid=null] - The record uuid to filter by.
+ * @param {string} [params.parentUuid=null] - The parent node uuid to filter by.
+ * @param {string} [params.nodeDefUuid=null] - The node definition uuid to filter by.
+ * @param {boolean} [params.draft=false] - Whether to fetch draft props or only published ones.
+ * @returns {string} - The select query.
+ */
+export function getSelect(params) {
+  const { uuid = null, recordUuid = null, parentUuid = null, nodeDefUuid = null, draft = false } = params
+
+  const _getPropsCombined = (table) => (draft ? `${table}.props || ${table}.props_draft` : `${table}.props`)
+
+  const propsTaxon = _getPropsCombined('t')
+  const propsVernacularName = _getPropsCombined('v')
+  const propsCategoryItem = _getPropsCombined('c')
+
+  const schemaSurvey = Schemata.getSchemaSurvey(this.surveyId)
```

Comment:
can you use here this.schema?

Reply:
the same file is used to generate the select for the result node table; in that case the schema is different.

Reply:
then add the table object models needed. (category_item, etc..)

Reply:
ok